projects/core/src/base-core.module.ts

```diff
@@ -32,6 +56,14 @@ export class BaseCoreModule {
   static forRoot(): ModuleWithProviders<BaseCoreModule> {
     return {
       ngModule: BaseCoreModule,
+      providers: [
+        {
+          provide: APP_INITIALIZER,
+          useFactory: moduleInitializersFactory,
+          deps: [LazyModulesService, [new Optional(), MODULE_INITIALIZER]],
+          multi: true,
+        },
+      ],
```

Comment:
We need to configure this APP_INITIALIZER ini a module we are sure that it will be present ini the app.
The `forRoot()` func in `BaseCoreModule` is one candidate.

Lets discuss to see if there is a better alternative.

Reply:
Perhaps we could encapsulate this better by creating a `LazyLoadingModule` in `projects/core/src/lazy-loading`.  `LazyLoadingModule` could define the APP_INITIALIZER and `BaseCoreModule` would then import `LazyLoadingModule`

Reply:
I created `LazyLoadingModule`.  It seemed like a better design.